avalanche/benchmarks/utils/detection_dataset.py

```diff
@@ -138,37 +140,6 @@ def targets_task_labels(self) -> DataAttribute[int]:
 ]
 
 
-@overload
-def make_detection_dataset(
-    dataset: SupervisedDetectionDataset,
-    *,
-    transform: Optional[XTransform] = None,
-    target_transform: Optional[YTransform] = None,
-    transform_groups: Optional[Mapping[str, TransformGroupDef]] = None,
-    initial_transform_group: Optional[str] = None,
-    task_labels: Optional[Union[int, Sequence[int]]] = None,
-    targets: Optional[Sequence[TTargetType]] = None,
-    collate_fn: Optional[Callable[[List], Any]] = None
-) -> SupervisedDetectionDataset:
-    ...
-
-
-@overload
-def make_detection_dataset(
-    dataset: SupportedDetectionDataset,
-    *,
-    transform: Optional[XTransform] = None,
-    target_transform: Optional[YTransform] = None,
-    transform_groups: Optional[Mapping[str, TransformGroupDef]] = None,
-    initial_transform_group: Optional[str] = None,
-    task_labels: Union[int, Sequence[int]],
-    targets: Sequence[TTargetType],
-    collate_fn: Optional[Callable[[List], Any]] = None
-) -> SupervisedDetectionDataset:
-    ...
-
-
-@overload
 def make_detection_dataset(
     dataset: SupportedDetectionDataset,
```

Comment:
do we need this? DetectionDataset should be enough. We have the others for classificaiton mostly to keep it compatible with the old API.

Reply:
It's also a matter of allowing:

- On the Avalanche side, to automatically fetch the `targets` from the dataset
- On the user side, to set the targets and task labels

The alternative option is setting the targets and task labels through `data_attributes`, but that seems an advanced topic.